target/linux/ramips/dts/mt7628an_tuoshi_lt22m.dts

```diff
@@ -0,0 +1,190 @@
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
+
+/ {
+	compatible = "tuoshi,lt22m", "mediatek,mt7628an-soc";
+	model = "Tuoshi LT22M 4G Wireless Router";
+
+	leds {
+		compatible = "gpio-leds";
+		led-wlan {
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_WLAN;
+			linux,default-trigger = "phy0tpt";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+
+		led-lan-1 {
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_LAN;
+			function-enumerator = <1>;
+			gpios = <&gpio 43 GPIO_ACTIVE_LOW>;
+		};
+
+		led-lan-2 {
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_LAN;
+			function-enumerator = <2>;
+			gpios = <&gpio 42 GPIO_ACTIVE_LOW>;
+		};
+
+		led-2g {
+			color = <LED_COLOR_ID_BLUE>;
+			function = "wwan";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+
+		led-3g {
+			color = <LED_COLOR_ID_BLUE>;
+			function = "wwan-3g";
```

Comment:
I think it's better to keep `function` as `wwan` and use `function-enumerator` for 3g/4g designations.

Reply:
I disagree. Lan1...Lan2 has no meaning, it's just a number. wwan, 3g, 4g has a meaning that get lost when using enumerators. Lan with function enumerators also results in devices that matches the labels on the device. Wwan with enumerators results in something unrelated to the labels.

Reply:
I tried to find examples for either variant but there seems to be none. Other devices just use "3g" or "4g" as names.

Reply:
> I think it's better to keep `function` as `wwan` and use `function-enumerator` for 3g/4g designations.

+1. `function-enumerator = 3` for 3g, `function-enumerator = 4` for lte. etc.

Reply:
I really don't like that. Mobile technologies are not a numeric enumeration. It asks for inconsistencies.

I have checked WIFI:
target/linux/ramips/dts/mt7621_rostelecom_rt-fe-1a.dts - Enumerators 50 (for 5 GHz) and 24 (for 2.4 GHz)
target/linux/ramips/dts/mt7621_iptime_a3004t.dts - Enumerators 0 and 1 (no context which is which)
target/linux/ramips/dts/mt7621_tplink_archer-x6-v3.dtsi - Enumerators 5 and 2 (which seems the most logical but is semantically wrong since it's not 2 GHz but 2.4 GHz)

IMO it's completely wrong usage of the enumerators.

Reply:
I have checked the docs: https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/common.yaml

```
  function:
    description:
      LED function. Use one of the LED_FUNCTION_* prefixed definitions
      from the header include/dt-bindings/leds/common.h. If there is no
      matching LED_FUNCTION available, add a new one.
    $ref: /schemas/types.yaml#/definitions/string

  function-enumerator:
    description:
      Integer to be used when more than one instance of the same function is
      needed, differing only with an ordinal number.
    $ref: /schemas/types.yaml#/definitions/uint32
```

Seems very clear to me that WiFi and Mobile types are _not_ the "same function, differing only with an ordinal number". And it's also clearly stated that a new LED_FUNCTION_ name should be added if one is missing for a particular function.

Reply:
WWAN function (or modem, or whatever they decide to call it) is indeed missing and is not the same as WLAN (or BT, or ZigBee/Matter, ...)
But I don't think wwan-4g is a reasonable function (analogous to having different functions for USB2 and USB3).